src/components/OpenSeadragonViewer.js

```diff
@@ -44,7 +44,14 @@ export class OpenSeadragonViewer extends Component {
     const { tileSources } = this.props;
     if (!this.tileSourcesMatch(prevProps.tileSources)) {
       this.viewer.close();
-      tileSources.forEach(tileSource => this.addTileSource(tileSource));
+      Promise.all(
+        tileSources.map(tileSource => this.addTileSource(tileSource)),
+      ).then(() => {
+        // An incredibly naive way to evaluate this for now.
```

Comment:
Does this warrant a `FIXME` or `TODO` prefix?

Reply:
Sure! Thanks for the suggestion. I've pushed up something adding the TODO and more context.